plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/S3Repository.java

```diff
@@ -146,6 +152,44 @@ class S3Repository extends MeteredBlobStoreRepository {
         MAX_PART_SIZE_USING_MULTIPART
     );
 
+    /**
+     * Minimum part size for parallel multipart uploads
+     */
+    static final Setting<ByteSizeValue> PARALLEL_MULTIPART_UPLOAD_MINIMUM_PART_SIZE_SETTING = Setting.byteSizeSetting(
+        "parallel_multipart_upload.minimum_part_size",
+        DEFAULT_MULTIPART_UPLOAD_MINIMUM_PART_SIZE,
+        MIN_PART_SIZE_USING_MULTIPART,
+        MAX_PART_SIZE_USING_MULTIPART,
+        Setting.Property.NodeScope
+    );
```

Comment:
How do we ensure these are optimal sizes like 8MB or 16MB

Reply:
Took this from transfer manager https://github.com/aws/aws-sdk-java/blob/master/aws-java-sdk-s3/src/main/java/com/amazonaws/services/s3/transfer/TransferManagerConfiguration.java#L38